apps/cmp.c

```diff
@@ -1509,10 +1509,23 @@ static int setup_request_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
         CMP_warn("no -subject given; no -csr or -oldcert or -cert available for fallback");
 
     if (opt_cmd == CMP_IR || opt_cmd == CMP_CR || opt_cmd == CMP_KUR) {
-        if (opt_newkey == NULL && opt_key == NULL && opt_csr == NULL) {
-            CMP_err("missing -newkey (or -key) to be certified and no -csr given");
+        if (opt_newkey == NULL
+            && opt_key == NULL && opt_csr == NULL && opt_oldcert == NULL) {
+            CMP_err("missing -newkey (or -key) to be certified and no -csr, -oldcert, or -cert given for fallback public key");
             return 0;
         }
+        if (opt_newkey == NULL
+            && opt_popo != OSSL_CRMF_POPO_NONE
+            && opt_popo != OSSL_CRMF_POPO_RAVERIFIED) {
+            if (opt_csr != NULL) {
+                CMP_err("no -newkey option given with private key for POPO, and any fallback -key option hidden by -csr");
```

Comment:
This is slightly confusing, specifically the "hidden" term. Is there a better way to rephrase it? (e.g. `-key option ignored with -csr option`)?

Reply:
The `-key` option is not entirely ignored with `-csr`, but I re-phrased that part of the error message and made it hopefully more clear and made it dependent on the presence of `-key`.